core-api-library/library/src/main/java/net/gini/android/core/api/DocumentRepository.kt

```diff
@@ -29,6 +31,13 @@ abstract class DocumentRepository<E: ExtractionsContainer>(
     private val giniApiType: GiniApiType
 ) {
 
+    /*
+    We need mutex lock because otherwise when the user upload multiple documents at first use of the app, the app
+    creates multiple users (equal to number of documents) and we will get a server error (the document does not belong to the user)!
+    we are using this Mutex in the withAccessToken() function so that it affect the whole application
```

Comment:
[Suggestion]
```suggestion
    We are using a Mutex to prevent coroutines from retrieving access tokens in parallel. This way even when multiple uploads are started only one user is created.
```